python/foglamp/tasks/north/sending_process.py

```diff
@@ -303,67 +220,60 @@ class SendingProcess:
             "type": "string",
-            "default": _DATA_SOURCE_READINGS
+            "default": "readings"
         },
         "blockSize": {
             "description": "Bytes to send in each transmission",
             "type": "integer",
             "default": "500"
         },
+        "sleepInterval": {
+            "description": "Time in seconds to wait between duration checks",
+            "type": "integer",
+            "default": "1"
+        },
+        'plugin': {
+            'description': 'The name of the translator to use to translate the readings into the output format and send them.',
+            'type': 'string',
+            'default': 'omf'
+        },
+        'filter': {
+            'description': 'Data filtering rules to apply to the data that has been pulled from the storage layer.',
+            'type': 'string',
+            'default': 'None'
+        },
+        'aggregate': {
+            'description': 'Aggregation rules to apply to the data',
+            'type': 'string',
+            'default': 'None'
+        },
         "memory_buffer_size": {
             "description": "Number of elements of blockSize size to be buffered in memory",
             "type": "integer",
             "default": "10"
         },
-        "north": {
-            "description": "Name of the north plugin to use to translate readings "
-                           "into the output format and send them",
-            "type": "string",
-            "default": "omf"
-        },
-        "stream_id": {
-            "description": "Stream ID",
-            "type": "integer",
-            "default": "1"
-        }
-
     }
 
     def __init__(self, loop=None):
-        """
-        Args:
-            _mgt_name: Unique name that represents the microservice
-            _mgt_port: Dynamic port of the management API - Used by the Storage layer
-            _mgt_address: IP address of the server for the management API - Used by the Storage layer
-        Returns:
-        Raises:
-        """
+        super().__init__()
 
-        # Initialize class attributes
         if not SendingProcess._logger:
             SendingProcess._logger = _LOGGER
-
-        # Configurations retrieved from the Configuration Manager
         self._config = {
             'enable': self._CONFIG_DEFAULT['enable']['default'],
             'duration': int(self._CONFIG_DEFAULT['duration']['default']),
             'source': self._CONFIG_DEFAULT['source']['default'],
             'blockSize': int(self._CONFIG_DEFAULT['blockSize']['default']),
-            'memory_buffer_size': int(self._CONFIG_DEFAULT['memory_buffer_size']['default']),
             'sleepInterval': float(self._CONFIG_DEFAULT['sleepInterval']['default']),
-            'north': self._CONFIG_DEFAULT['north']['default'],
+            'plugin': self._CONFIG_DEFAULT['plugin']['default'],
+            'filter': self._CONFIG_DEFAULT['filter']['default'],
```

Comment:
applyFilter and filterRule, I think as of today its applied on each north plugin